Sources/Client/Client/ClientLogger.swift

```diff
@@ -125,18 +129,27 @@ public final class ClientLogger {
         }
     }
     
+    /// Controls whether to display icons in logs
+    /// Only valid when default `ClientLogger.logger` block is used, if you've overridden that, this is not valid.
+    public static var iconEnabled = true
+    
     /// A customizable logger block.
     /// By default error messages will print to the console, but you can customize it to use own logger.
     ///
     /// - Parameters:
     ///     - icon: a small icon string like a tag for messages, e.g. 🦄
     ///     - dateAndTime: a formatted string of date and time, could be empty.
+    ///     - level: Log level
     ///     - message: a message.
-    public static var logger: (_ icon: String, _ dateTime: String, _ message: String) -> Void = {
-        if $1.isEmpty || DateFormatter.log == nil {
-            print($0, $2)
+    public static var logger: (String, String, Level, String) -> Void = { icon, dateTime, level, message in
```

Comment:
This can be done in a non-breaking way by providing a different version of the logger variable.

Reply:
We can't overload the name since this is a closure (I've tried)
Do you suggest adding a new logger block (`loggerWithLevel` or something) and keeping the old one as is?
I agree this should be done in a non-breaking way but I couldn't find the solution.. No default argument for closure, no overloading....

Reply:
I see. Having another variable is probably the only option then. We need to prepare something more robust for v3.

Reply:
Another variable is not an issue.
Naming that variable, on the other hand.....

Reply:
`levelAwareLogger`?

Reply:
Updated! Non-breaking and compiler-assistant way